dcurve/src/internal/fetch.ts

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+import { Contract } from 'ethers';
+import { Contribution, GrantRoundContributions, GrantRoundFetchArgs } from '../../src/types';
+import { abi as GRANT_ROUND_MANAGER_ABI } from '@dgrants/contracts/artifacts/contracts/GrantRoundManager.sol/GrantRoundManager.json';
+import { abi as GRANT_REGISTRY_ABI } from '@dgrants/contracts/artifacts/contracts/GrantRegistry.sol/GrantRegistry.json';
+
+const ERC20_ABI = ['function balanceOf(address) view returns (uint)'];
+
+/**
+ * Returns GrantRoundContributions describing the round and any contributions
+ * @param {GrantRoundFetchArgs}
+ */
+export const fetch = async (args: GrantRoundFetchArgs) => {
+  // fetch grantRoundManager
+  const roundManager = new Contract(args.grantRoundManager, GRANT_ROUND_MANAGER_ABI, args.provider);
+  // fetch the grantList
+  const registry = new Contract(args.grantRegistry, GRANT_REGISTRY_ABI, args.provider);
+
+  // set-up
+  const values = await Promise.all([
+    registry.getAllGrants(),
```

Comment:
Just a note this may fail if the number of grants is too large to return in a single call, in which case this will need to be chunked with `getGrants()`. Doesn't matter for now, but something to be aware as a potential future source of error

Reply:
Left a todo! will create an issue for this as well later

Reply:
Being tracked in https://github.com/dcgtc/dgrants/issues/125